src/main/java/org/folio/event/handler/ItemCreateAsyncRecordHandler.java

```diff
@@ -0,0 +1,117 @@
+      log.debug("ItemCreateAsyncRecordHandler::handle, kafkaRecord={}", kafkaConsumerRecord.value());
+    }
+
+    var payload = new JsonObject(kafkaConsumerRecord.value());
+
+    String eventType = payload.getString("type");
+    if (!Objects.equals(eventType, DomainEventPayloadType.CREATE.name())) {
+      log.info("processItemCreateEvent:: unsupported event type: {}", eventType);
+      return Future.succeededFuture();
+    }
+
+    if (!validatePayload(payload)) {
+      log.warn("processItemCreateEvent:: payload validation failed");
+      return Future.succeededFuture();
+    }
+
+    try {
+      var tenantId = extractTenantFromHeaders(kafkaConsumerRecord.headers());
+      var dbClient = new DBClient(getVertx(), tenantId);
+      return processItemCreationEvent(payload, dbClient)
+        .onSuccess(v -> log.info("ItemCreateAsyncRecordHandler::handle, event processed successfully"))
+        .onFailure(t -> log.error("Failed to process event: {}", kafkaConsumerRecord.value(), t))
+        .map(kafkaConsumerRecord.key());
+    } catch (Exception e) {
+      log.error("Failed to process export history kafka record, kafkaRecord={}", kafkaConsumerRecord, e);
+      return Future.failedFuture(e);
+    }
+  }
+
+  private Future<Void> processItemCreationEvent(JsonObject payload, DBClient dbClient) {
+    var tenantId = payload.getString("tenant");
+    var itemObject = payload.getJsonObject("new");
+    var itemId = itemObject.getString("id");
+
+    return pieceService.getPiecesByItemId(itemId, dbClient)
+      .compose(pieces -> updatePieces(pieces, itemObject, tenantId, dbClient));
+  }
+
+  private Future<Void> updatePieces(List<Piece> pieces, JsonObject itemObject, String tenantId, DBClient client) {
+    if (CollectionUtils.isEmpty(pieces)) {
+      String itemId = itemObject.getString("id");
+      log.info("updatePieces:: no pieces to update found, nothing to update for item={}, tenant={}",
+        itemId, tenantId);
+      return Future.succeededFuture();
+    }
+
+    var holdingId = itemObject.getString("holdingsRecordId");
+    var updateRequiredPieces = filterPiecesToUpdate(pieces, holdingId, tenantId);
+    updatePieceFields(updateRequiredPieces, holdingId, tenantId);
+    return pieceService.updatePieces(updateRequiredPieces, client);
+  }
+
+  private List<Piece> filterPiecesToUpdate(List<Piece> pieces, String holdingId, String tenantId) {
+    return pieces.stream()
+      .filter(piece ->
+        // filter out pieces that already have the same tenantId and (same holdingId or existing locationId)
+        ObjectUtils.notEqual(piece.getReceivingTenantId(), tenantId)
+          && Objects.isNull(piece.getLocationId())
+          && ObjectUtils.notEqual(piece.getHoldingId(), holdingId))
+      .toList();
+  }
+
+  private void updatePieceFields(List<Piece> pieces, String holdingId, String tenantId) {
+    pieces.forEach(piece -> {
+      piece.setReceivingTenantId(tenantId);
+      if (Objects.isNull(piece.getLocationId())) {
```

Comment:
we can skip this if statement t oset holdingId, because for Item Create event holdingId could not be changed. It can be changed for Item Update event

Reply:
I prefer extra checks without dependent on other modules. The reason is that the logic can be changed, in this case this part is working without know main logic of process and can fix it. in my opinion It is useful